lib/EnzymeCore/src/rules.jl

```diff
@@ -26,6 +37,12 @@ const ConfigWidth{Width} = Config{<:Any,<:Any, Width}
 @inline width(::Config{<:Any, <:Any, Width}) where Width = Width
 @inline overwritten(::Config{<:Any, <:Any, <:Any, Overwritten}) where Overwritten = Overwritten
 
+"""
+    AugmentedReturn(primal, shadow, tape)
+
+Augment the primal return value of a function with its shadow, as well as any additional saved information 
```

Comment:
Don't really understand tape, so was vague

Reply:
perhaps "any additional information needed to correctly compute the reverse pass. Unless specified by the config that a variable is not overwritten, rules must assume any arrays/data structures/etc are overwritten between the forward and the reverse pass. Any floats or variables passed by value are always preserved as is (as are the arrays themselves, just not necessarily the values in the array)."